static/app/components/profiling/breadcrumb.tsx

```diff
@@ -0,0 +1,70 @@
+import {useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import Breadcrumbs, {Crumb} from 'sentry/components/breadcrumbs';
+import {t} from 'sentry/locale';
+import {Organization} from 'sentry/types';
+import {
+  generateFlamegraphRoute,
+  generateProfilingRoute,
+} from 'sentry/views/profiling/routes';
+
+type ProfilingTrail = {
+  type: 'profiling';
+};
+
+type FlamegraphTrail = {
+  payload: {
+    interactionName: string;
+    profileId: string;
+    projectSlug: string;
+  };
+  type: 'flamegraph';
+};
+
+type Trail = ProfilingTrail | FlamegraphTrail;
+
+interface BreadcrumbProps {
+  organization: Organization;
+  trails: Trail[];
+}
+
+function Breadcrumb({organization, trails}: BreadcrumbProps) {
+  const crumbs = useMemo(
+    () => trails.map(trail => trailToCrumb(trail, {organization})),
+    [trails]
+  );
+  return <StyledBreadcrumbs crumbs={crumbs} />;
+}
+
+function trailToCrumb(trail: Trail, {organization}: {organization: Organization}): Crumb {
+  const trailType = trail.type;
+  switch (trailType) {
```

Comment:
```suggestion
  switch (trail.type) {
```
is descriptive enough imo